fomantic/Fomantic-UI
# feat(site): offline bundled lato 2.0 and systemfonts, no googleimport by default

## Description
This PR bundles the recent Lato 2.0 Font. It uses the same styles as the previous googleFonts Import does.

This fixes possible GDPR issues as the default theme now disables googlefont import.

I implemented a new variable `@fonts` which allows for a totally free description of font faces to be imported if needed. Each declared property will be used, so one can customize them as needed by that single variable.

Disabling the whole import can be done via  `@importFont: false` or setting `@font: {}` or setting `@fontName: ''` just as one likes.
The new variables for unicode or font properties are done to simplify the default import and of course if one just wants to adjust them without redeclaring the whole `@font` variable mixin.

I also added `@supportIE` which allows to disable the woff-font import, as all other supported browsers support woff2 already.
> This new variable and also the icon font-face will be adjusted accordingly in a separate PR... in case you might ask/wonder 😉 

The default theme uses Lato-Latin 2.0  now which is smaller in size but includes the most important glyphs for European Latin-based languages which were missing in Lato v1 (via googlefonts) like in "Řeřicha"
It also uses Lato with Greek+Cyrillic+Extended 2.0 via `unicode-range` property which makes sure the font is only loaded when related characters are used inside the page.
If no character matches at all, the system fonts will be used as fallback.

I created the unicode ranges by analyzing the fonts character ranges using https://github.com/fontforge/fontforge 

I also added an optional theme to adjust the font via a change in theme.config:

- systemfont (which completely disables any font import and uses the platform system fonts only instead AND adjusts the bold setting so the overall visual appearance is very similar) See comments below for more info.

The used Lato fonts are available free of charge under the SIL Open Font License from http://www.latofonts.com/

## Testcase
Build FUI :)

### To test System Fonts
- Adjust `theme.config`:
```less
@site       : 'systemfont';
```
- Build FUI
- Use some latin as well as cyrillic and greek characters on a webpage
- inspect the character sections to see which font is used

## Screenshots
![lato2](https://user-images.githubusercontent.com/18379884/168491993-0efdbdcc-f76a-46e0-bde6-023cfd75d2c7.gif)

|Lato 2.0|Lato 1.0 (until FUI 2.8.8)| systemfont|
|-|-|-|
![image](https://user-images.githubusercontent.com/18379884/166122584-74bf7196-66ac-42d2-a4de-0f1dacd62e25.png)|![image](https://user-images.githubusercontent.com/18379884/166122653-d60aca63-8994-4d3e-8c65-1b58d59404f2.png)|![image](https://user-images.githubusercontent.com/18379884/166122684-8f8b3d51-43fa-4979-b0d2-b35d0a586bd3.png)|

## Closes
#367 
#2355 
https://github.com/Semantic-Org/Semantic-UI/issues/7075
